internal/api/http.go

```diff
@@ -167,11 +170,11 @@ func newHeaderRoundTripper(host string, authToken string, headers map[string]str
 			}
 		}
 	}
-	if _, ok := headers[authorization]; !ok && authToken != "" {
-		headers[authorization] = fmt.Sprintf("token %s", authToken)
-	}
 	if _, ok := headers[timeZone]; !ok {
-		headers[timeZone] = currentTimeZone()
+		tz := currentTimeZone()
+		if tz != "" {
+			headers[timeZone] = tz
```

Comment:
This alone looks like it's sufficient to solve the original issue (sending the blank Time-Zone header). How come additional changes to the functionality are needed?

Reply:
The reason that is not sufficient to solve the bug is because [this](https://github.com/cli/cli/blob/02881b4783db05be2249c27f36130e3ef82daa9a/api/client.go#L257) line is explicitly telling `go-gh` not to resolve the "Time-Zone" header and leave it set to an empty string. We could detect that in `go-gh` and remove the "Time-Zone" header if it set to an empty string but that felt off since we have been following the rule that any user supplied values are not modified. Adding the `SkipResolution` options feels like a cleaner solution and also addresses the confusion you correctly identified [here](https://github.com/cli/cli/pull/5935#discussion_r921065970).

Reply:
Fair enough, but if there is going to be an additional config option, maybe it could be better documented? Right now I don't think it's clear what "resolution" is, since the user probably isn't familiar with the implementation of go-gh. Since the configuration option only affects whether default headers will be added, maybe the option could be called `skipDefaultHeaders`?

Reply:
Great suggestions, that is a much more descriptive name.